src/Analyser/MutatingScope.php

```diff
@@ -1005,6 +1005,29 @@ private function resolveType(Expr $node): Type
 				return $leftStringType->append($rightStringType);
 			}
 
+			if ($leftStringType instanceof UnionType && $rightStringType instanceof ConstantStringType) {
```

Comment:
For my use case even

```suggestion
			if ($leftStringType instanceof UnionType && $rightStringType->isLiteralString())->yes()) {
```

should be enough and less performance intensive